src/test/java/org/sagebionetworks/web/unitclient/widget/entity/act/ApproveUserAccessModalTest.java

```diff
@@ -0,0 +1,300 @@
+		verify(mockProgressWidget).startAndTrackJob(anyString(), anyBoolean(), any(AsynchType.class), any(QueryBundleRequest.class), phCaptor.capture());
+		phCaptor.getValue().onCancel();
+		verify(mockSynAlert).showError("Query cancelled");
+	}
+	
+	@Test
+	public void testLoadEmailMessageOnComplete() {
+		dialog.configure(actList, entityBundle);
+		verify(mockProgressWidget).startAndTrackJob(anyString(), anyBoolean(), any(AsynchType.class), any(QueryBundleRequest.class), phCaptor.capture());
+		phCaptor.getValue().onComplete(qrb);
+		verify(mockView).finishLoadingEmail();
+	}
+	
+	@Test
+	public void testOnStateSelected() {
+		ACTAccessRequirement ar = actList.get(0);
+		String num = Long.toString(ar.getId());
+		String text = GovernanceServiceHelper.getAccessRequirementText(ar);
+		dialog.configure(actList, entityBundle);
+		dialog.onStateSelected(num);
+		verify(mockView, times(2)).setAccessRequirement(eq(num), eq(text));
+	}
+	
+	@Test
+	public void testOnSubmitNoUserSelected() {
+		dialog.configure(actList, entityBundle);
+		dialog.onSubmit();
+		verify(mockSynAlert).showError(eq("You must select a user to approve"));
+	}
+	
+	@Test
+	public void testOnSubmitNoMessage() {
+		dialog.configure(actList, entityBundle);
+		when(mockGlobalApplicationState.getSynapseProperty(anyString())).thenReturn(null);
+		dialog.onUserSelected(mockUser);
+		dialog.onSubmit();
+		verify(mockSynAlert, times(0)).showError(eq("You must select a user to approve"));
+		verify(mockSynAlert).showError("An error was encountered while loading the email message body");
+	}
+	
+	@Test
+	public void testOnSubmitNullAccessReq() {
+		dialog.configure(actList, entityBundle);
+		
+		verify(mockProgressWidget).startAndTrackJob(anyString(), anyBoolean(), any(AsynchType.class), any(QueryBundleRequest.class), phCaptor.capture());
+		phCaptor.getValue().onComplete(qrb);
+		
+		dialog.onUserSelected(mockUser);
+		dialog.onSubmit();
+		verify(mockSynAlert, times(0)).showError(eq("You must select a user to approve"));
+		verify(mockSynAlert, times(0)).showError("An error was encountered while loading the email message body");
+		verify(mockView).getAccessRequirement();
+		verify(mockView).setApproveProcessing(true);
+	}
+	
+	@Test
+	public void testOnSubmitOnFailure() {
+		dialog.configure(actList, entityBundle);
+		
+		verify(mockProgressWidget).startAndTrackJob(anyString(), anyBoolean(), any(AsynchType.class), any(QueryBundleRequest.class), phCaptor.capture());
+		phCaptor.getValue().onComplete(qrb);
+		
+		dialog.onUserSelected(mockUser);
+		dialog.onSubmit();
+		verify(mockSynAlert, times(0)).showError(eq("You must select a user to approve"));
+		verify(mockSynAlert, times(0)).showError("An error was encountered while loading the email message body");
+		verify(mockView).getAccessRequirement();
+		verify(mockView).setApproveProcessing(true);
+		
+		verify(mockSynapseClient).createAccessApproval(any(ACTAccessApproval.class), aaCaptor.capture());
```

Comment:
this works, but for convenience you may want to respond with a success by default (use AsyncMockStubber.callSuccessWith() in Before.